src/Symfony/Component/Form/Extension/Core/Type/ChoiceType.php

```diff
@@ -404,10 +404,14 @@ private function createChoiceList(array $options)
 
     private function createChoiceListView(ChoiceListInterface $choiceList, array $options)
     {
+        if (false === $label = $options['choice_label']) {
+            $label = static function () { return false; };
```

Comment:
this means `false` is not passed to the choiceListFactory anymore. so a custom implementation would not get the expected value anymore. this is the same BC break as in #32955

Reply:
The difference is that with this solution option `label` remains false if it it was set to `false`. User can still do `false !== $form->getConfig()->getOption('choice_label')` or `$resolver->setNormalizer('custom_option', fn($options) => false === $options['label'])`.

At the same time I do pass only callables to `createChoiceListView`.

Reply:
While with your solution the resolved `$option['label']` is never `false` anymore, it's always a callable, so it changes it's range of values without any alert.

Reply:
or instead of this premature normalization here, we could add a new hidden argument `/*, $triggerDeprecation = true */` to the `createView` method to skip the warning on demand, thus custom `choiceListFactory` will continue receiving the `false' value. We have many examples of the alternative in other places.